matlab/src/matlab/+arrow/+tabular/+internal/makeValidVariableNames.m

```diff
@@ -0,0 +1,36 @@
+%MAKEVALIDVARIABLENAMES Makes valid table variable names.
+
+% Licensed to the Apache Software Foundation (ASF) under one or more
+% contributor license agreements.  See the NOTICE file distributed with
+% this work for additional information regarding copyright ownership.
+% The ASF licenses this file to you under the Apache License, Version
+% 2.0 (the "License"); you may not use this file except in compliance
+% with the License.  You may obtain a copy of the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+function [varnames, modified] = makeValidVariableNames(varnames)
+    arguments
+        varnames(1, :) string
+    end
+
+    reservedNames = ["Properties" "VariableNames" "RowNames", ":"];
```

Comment:
It might be nice to use commas between all the elements here (currently there is only a comma after "RowNames").